test/e2e/component_operator_test.go

```diff
@@ -0,0 +1,88 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	"github.com/redhat-developer/devopsconsole-operator/pkg/apis"
+	componentsv1alpha1 "github.com/redhat-developer/devopsconsole-operator/pkg/apis/devopsconsole/v1alpha1"
+
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+)
+
+// ComponentTest does e2e test as per operator-sdk documentation
+// https://github.com/operator-framework/operator-sdk/blob/cc7b175/doc/test-framework/writing-e2e-tests.md
+func TestComponent(t *testing.T) {
+	// Register types with framework scheme
+	componentList := &componentsv1alpha1.ComponentList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Component",
+			APIVersion: "devopsconsole.openshift.io/v1alpha1",
+		},
+	}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, componentList)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+
+	defer os.Unsetenv("TEST_NAMESPACE")
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	t.Log(fmt.Sprintf("namespace: %s", namespace))
+	// wait for component-operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "devopsconsole-operator", 1, retryInterval, timeout)
```

Comment:
The namespace here should be "operators", I believe

Reply:
@Avni-Sharma Indeed.
That should be the namespace in https://github.com/redhat-developer/devopsconsole-operator/pull/34/files#diff-153541cef4d3666ae354851a6f6ef7c2R8 
@tinakurian we should still have 2 namespaces:
- one where we deploy the operator (could be "operators" as suggested by Avni, on the one i've used devopsconsole)
- one where a use wants to deploy its component CR, as one requirement is to be able to make sure we can watch CR from _any_ namespace, I'd rather have 2 different one.

I think with the changed you did we're atm use only one namespace (which should work too) but to reflect reality better have 2 different one.

Reply:
@tinakurian after further thoughts, I think we better keep that first PR simple, pls clean the resources and then we merge that one. 
We can then add more tests.